src/super_gradients/training/pre_launch_callbacks/pre_launch_callbacks.py

```diff
@@ -104,39 +111,51 @@ def __call__(self, cfg: DictConfig) -> DictConfig:
         tmp_cfg.training_hyperparams.kill_ddp_pgroup_on_end = False
         tmp_cfg.pre_launch_callbacks_list = []
 
+        fastest_batch_time = np.inf
```

Comment:
I had to read couple time the logic to catch what is happening and when batch size is increased actually. I think it would be great to add to a docstring to this function a description of the logic and what are the exit criterions. 
Currently there are two return statements from the `while True` loop. Maybe it is possible to refactor the code to have exactly one return point?